[View Variables] Teleport Here

DeityLink opened the pull request:
Adds "Teleport Here" to the View Variables' drop down menu.
Teleports the mob/obj to where you are standing.
Similar to the Player Panel's "Get" button, however this one also works on objects.
Produces a "flash" like the one produced by admins spawning-in as Test Dummies.

![here](https://cloud.githubusercontent.com/assets/7573912/8872132/b1c34898-31fa-11e5-8d1c-75cf74528f2a.png)
![chair_tp](https://cloud.githubusercontent.com/assets/7573912/8872150/d2a8752e-31fa-11e5-8615-66c79d508d29.png)

PJB3005 commented:
:+1:

DeityLink commented:
As was requested by @clusterfack added a Delete option.
If the atom is a mob with a player in it, confirmation will be asked.

clusterfack commented on code/datums/datumvars.dm:
```diff
@@ -704,6 +709,46 @@ client
+		if(!check_rights(0))	return
+
+		var/atom/A = locate(href_list["teleport_here"])
+		if(!isobj(A) && !ismob(A))
```
You can have it typecast as atom/movable instead and check against that

clusterfack commented on code/datums/datumvars.dm:
```diff
@@ -704,6 +709,46 @@ client
+			O.loc = T
+		else if(ismob(A))
+			var/mob/M = A
+			M.loc = T
```
Forcemove is the standard now

clusterfack commented:
Thanks this'll be helpful

IratePirate commented:
The more behind-the-scenes admin tools like this you improve and add, the more potential the serb has for great adminbus. Good on you.

ComicIronic commented:
The flash should be optional, you should be able to delete anything (you already can with a verb) and you should cast as an atom/movable for teleport, rather than checking mob and obj.

clusterfack commented:
You can del anything with the right click verb yes (the del menu itself is too vague to be useful), but inside objects or ones found through VV-ing around cannot be very easily.

DeityLink commented:
@ComicIronic this option is mainly meant for adminbus when you want to move something to an area with players and with an effect to add some more credibility to an hypothetical "We teleported this item to your station -signed Nanotrasen" announcement.
If you don't want a flash you can just Varedit.

ComicIronic commented:
I support being able to del from the VV menu, but there's no reason to restrict it.

ComicIronic commented on code/datums/datumvars.dm:
```diff
@@ -267,6 +267,9 @@ client
 		if(ismob(D))
 			body += "<option value='?_src_=vars;mob_player_panel=\ref[D]'>Show player panel</option>"
 
+		if(istype(D,/atom))
```
/atom/movable

DeityLink replied:
k

d3athrow commented:
I don't like the flash being not optional

DeityLink commented:
You're the boss, here have some modularity:

* Added an option in the View Variables's drop down menu that lets admins teleport a mob/object to where they are standing.
* Admins can set the teleportation to be either flashy, with effects or even screen shaking, or stealthy with a fade-in. This preference is set in the Debug tab.
* Added another option that lets admins delete a mob/object from the View Variables menu.
* Added an admin spawnable "calm" Nar-Sie that by default doesn't move, doesn't consume atoms, and doesn't trigger Hell nor force a shuttle call. Mostly for debug and testing purposes.
* Plastic Flaps are now replaced by cult grilles by Nar-Sie (which are airtight when not broken).
* Fixed not being able to move through partially broken cult grilles

d3athrow commented:
Frivilous addition that can be done easily with direct/global narrate

DeityLink commented:
@d3athrow but now you can have the message appear at the exact time to teleport the atom in without having to juggle with clicks

d3athrow commented:
BRBRBRBRBRBRBRBRBR

PJB3005 commented:
>Plastic Flaps are now replaced by cult grilles by Nar-Sie (which are airtight). 

ATOMICITY

DeityLink commented:
but it's basically a bugfix! otherwise areas like cargo get depressurized if the shuttle isn't there.

ComicIronic commented:
When we say "bugfixes don't have to be atomic", that means you can bundle lots of fixes together or you can fix while you change something.

It doesn't mean you can jam whatever bugfix into whatever PR!

DeityLink commented:
You're right, still, we want PRs to be atomic so they can be more easily reviewed right, and this "bugfix" is only 3 lines long so, can you make an exception? :(

d3athrow commented:
The atomic rule as far as bugfixes goes if you bundle a fix with a feature PR then that fix wont get merged until the feature part is ready as well

DeityLink commented:
Well the feature part of this PR has been waiting for Final Decision for 3 days. 
I've added a Teleport To option as you requested on Steam.
